contracts/core/EntryPoint.sol

```diff
@@ -345,27 +374,23 @@ contract EntryPoint is IEntryPoint, StakeManager {
         MemoryUserOp memory mUserOp = opInfo.mUserOp;
         address sender = mUserOp.sender;
         _createSenderIfNeeded(opIndex, opInfo, op.initCode);
+        address paymaster = mUserOp.paymaster;
         if (aggregator == SIMULATE_FIND_AGGREGATOR) {
             numberMarker();
 
-            if (sender.code.length == 0) {
-                // it would revert anyway. but give a meaningful message
-                revert FailedOp(0, address(0), "AA20 account not deployed");
+            // solhint-disable-next-line no-empty-blocks
+            try this._simulateFindAggregator(sender, paymaster) {}
+            catch Error(string memory revertReason) {
+                revert FailedOp(opIndex, address(0), revertReason);
             }
-            if (mUserOp.paymaster != address(0) && mUserOp.paymaster.code.length == 0) {
-                // it would revert anyway. but give a meaningful message
-                revert FailedOp(0, address(0), "AA30 paymaster not deployed");
-            }
-            // during simulation, we don't use given aggregator,
-            // but query the account for its aggregator
-            try IAggregatedAccount(sender).getAggregator() returns (address userOpAggregator) {
-                aggregator = actualAggregator = userOpAggregator;
-            } catch {
-                aggregator = actualAggregator = address(0);
+            catch (bytes memory revertReason) {
+                if (revertReason.length != 32) {
```

Comment:
how that can be? revert is never just ABI-encoded string (that is, you expect it to start with (zero word, length word and then data). If this is a catch-all for unknown error, then don't `abi.decode` it, just cast it to string. This way, it is possible to parse the FailedOp to find the original error. with this decode, it will juust revert with no data)

Reply:
Fixed.